src/Toml/Codec/Combinators.hs

```diff
@@ -67,6 +67,7 @@ module Toml.Codec.Combinators
       -- * Combinators for Maps
     , map
     , tableMapCodec
+    , dynamicMapCodec
```

Comment:
I think we don't need `tableMapCodec`, since `dynamicMapCodec` is a more general and powerful function. So we can remove `tableMapCodec`, since it's only a special case. But it would be a good idea to keep a `TODO` near the `dynamicMapCodec` function about adding docs